File: python/primos.py
def es_primo(n):
    if n <= 1:
        return False
    if n <= 3:
        return True
    if n % 2 == 0 or n % 3 == 0:
        return False

    i = 5
    while i * i <= n:
        if n % i == 0 or n % (i + 2) == 0:
            return False
        i += 6

    return True

limite = int(input("Introduce un número límite para encontrar números primos: "))

print(f"Números primos hasta {limite}:")
for i in range(2, limite + 1):
    if es_primo(i):
        print(i, end=" ")
print()

File: C/primos.c
#include <iostream>
using namespace std;

bool esPrimo(int n) {
    if (n <= 1) return false;
    if (n <= 3) return true;
    if (n % 2 == 0 || n % 3 == 0) return false;

    for (int i = 5; i * i <= n; i += 6) {
        if (n % i == 0 || n % (i + 2) == 0) return false;
    }

    return true;
}

int main() {
    int limite;
    cout << "Introduce un número límite para encontrar números primos: ";
    cin >> limite;

    cout << "Números primos hasta " << limite << ":" << endl;
    for (int i = 2; i <= limite; i++) {
        if (esPrimo(i)) {
            cout << i << " ";
        }
    }
    cout << endl;

    return 0;
}
